src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -203,6 +225,39 @@ public HttpResponse doFinishLogin(StaplerRequest request, StaplerResponse respon
     return HttpResponses.redirectTo(redirectUrl);
   }
 
+  /**
+   * Extract a usable Username from the samlProfile object.
+   * @param saml2Profile
+   * @return
+   */
+  private String getUsernameFromProfile(Saml2Profile saml2Profile) {
+    if (usernameAttributeName != null) {
+      Object attribute = saml2Profile.getAttribute(usernameAttributeName);
+      if (attribute instanceof String) {
+        return (String) attribute;
+      }
+      if (attribute instanceof List) {
+        return (String) ((List<?>)attribute).get(0);
+      }
+      LOG.log(Level.SEVERE, "Unable to get username from Saml Profile {0}", saml2Profile);
+      throw new Error("Attribute "+usernameAttributeName+" contains no usable username");
```

Comment:
In other places of this file for similar errors IllegalStateException class is used. Should not the IllegalStateException be used here as well?